.github/workflows/ci.yml

```diff
@@ -0,0 +1,130 @@
+        submodules: recursive
+
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: |
+          .yarn/cache
+          .pnp.js
+        key: yarn-deps-v1-${{ hashFiles('yarn.lock') }}
+        restore-keys: |
+          yarn-deps-v1-
+
+    - name: Install dependencies
+      run: yarn --immutable
+
+    - name: Lint code
+      run: yarn test:lint
+
+    - name: Run Flow
+      run: yarn test:flow
+      if: always()
+
+    - name: Run Jest tests
+      run: yarn test:jest --coverage
+      if: always()
+
+    - uses: codecov/codecov-action@v1
+
+  screenshotter:
+    runs-on: ubuntu-latest
+    if: |
+      !contains(toJSON(github.event.commits.*.message), '[skip ci]') &&
+      !contains(toJSON(github.event.commits.*.message), '[ci skip]')
+
+    strategy:
+      matrix:
+        browser: [chrome, firefox, safari]
+        include:
+        - browser: chrome
+          services:
+            selenium:
+              image: selenium/standalone-chrome:3.141.59-20200525
+              ports:
+              - 4444:4444
+        - browser: firefox
+          services:
+            selenium:
+              image: selenium/standalone-firefox:3.141.59-20200525
+              ports:
+              - 4444:4444
+        - browser: safari
+          services: { }
+          browserstack:
+            browserName: Safari
+            browser_version: "13.1"
+            os: OS X
+            os_version: Catalina
+          
+      fail-fast: false
+
+    services: ${{ matrix.services }}
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: |
+          .yarn/cache
+          .pnp.js
+        key: yarn-deps-v1-${{ hashFiles('yarn.lock') }}
+        restore-keys: |
+          yarn-deps-v1-
+
+    - name: Install dependencies
+      run: yarn --immutable
+
+    - name: Verify screenshots and generate diffs and new screenshots
+      run: yarn node dockers/screenshotter/screenshotter.js -b ${{ matrix.browser }} -c ${{ job.services.selenium.id }} --verify --diff --new
+      if: matrix.services.selenium
+
+    - name: Verify screenshots and generate diffs and new screenshots
+      run: yarn node dockers/screenshotter/screenshotter.js -b safari --verify --diff --new --browserstack --selenium-capabilities '${{ toJson(matrix.browserstack) }}'
+      if: matrix.browserstack && github.repository_owner == 'KaTeX'
+      env:
+        BROWSERSTACK_USER: ${{ secrets.BROWSERSTACK_USER }}
+        BROWSERSTACK_ACCESS_KEY: ${{ secrets.BROWSERSTACK_ACCESS_KEY }}
+
+    - uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: new
+        path: test/screenshotter/new
```

Comment:
I think it'd be good to have one commit in this PR that temporarily introduces a change so that this can be tested.

Reply:
@kevinbarabash I've opened #2418.

Reply:
You're right. Artifacts with the same name from different jobs overwrite each other. I'll update the PR.